accession-commons-monotonic-generator-jpa/src/test/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicMonotonicAccessioningWithAlternateRangesTest.java

```diff
@@ -156,49 +159,44 @@ public void testAlternateRangesWithDifferentGenerators() throws AccessionCouldNo
         assertEquals(9, evaAccessions.size());
         //New Block from different instance have not jumped as still blocks are available before interleaving point
         assertNotEquals(80, evaAccessions.get(0).getAccession().longValue());
-        assertEquals(50, evaAccessions.get(0).getAccession().longValue());
-        assertEquals(58, evaAccessions.get(8).getAccession().longValue());
-        assertEquals(1, getUncompletedBlocksByCategoryIdAndApplicationInstanceIdOrderByEndAsc(contiguousIdBlockService.getRepository(), categoryId, instanceId2).size());
+        assertEquals(49, evaAccessions.get(0).getAccession().longValue());
+        assertEquals(57, evaAccessions.get(8).getAccession().longValue());
+        assertEquals(1, getAllUncompletedBlocksForCategoryId(contiguousIdBlockRepository, categoryId).size());
```

Comment:
To be honest, I don't really understand the block interleaving and how it's impacted by this PR... I think it's probably fine though.

Reply:
This PR does not have any impact on the block interleaving logic. 
However, it is affecting this test case as this was written with the notion that an AccessionGenerator will try to recover and use accession from an uncompleted block only if it belongs to the same instance. 
But with the updated logic, that is no longer the case and an AccessionGenerator tries to recover and use accessions from all the uncompleted blocks it can find in DB without considering their instance.